### ISSUE TITLE
image registration failed

### ISSUE THREAD

**Issue by qiuyixmm** [STOmics/SAW#128]
Hello, 
The SAW (version: 7.0) pipeline stopped at the step of image registration and threw an error as shown below:
<img width="1918" alt="image" src="https://github.com/STOmics/SAW/assets/30545631/c085bba9-1b87-4106-b6fe-5dd6e4f66aaf">

Please help me at your convenience. 
Thanks!

**Comment by Jinghao-Chen** [comment]
Hi, this may be because your 02.count/SN.raw.gef, that is, the gene expression matrix file is incomplete. You could confirm whether the SAW count step was completed successfully, or upload all the log files to us.

**Comment by qiuyixmm** [comment]
Yes, you are right! There are no "transcript" terms in the 3rd column of my annotation file in GTF format.  For that, I use a new annotation file in GFF format, which have "transcript" and "mRNA" terms in its 3rd column, like this:

<img width="1914" alt="image" src="https://github.com/STOmics/SAW/assets/30545631/d74975b4-eb96-41b2-a53b-25ee978ffb66">

I have some small questions:
1. For the counting process, the reads that are mapped to exons of both "transcript" and "mRNA" terms would be calculated when we use a GFF file as an annotation file?
2. I notice that some genes only have "exon" terms, and don't include "transcript" and "mRNA" terms. The read counts for these genes would be reported in the final gene expression matrix?
3.  For the annotation file derived from ensemble database,  the gene_id (ENSGALG00000016906) or gene_name(SPRY2) would be used in the final gene expression matrix?

<img width="1919" alt="image" src="https://github.com/STOmics/SAW/assets/30545631/e815d197-ba73-45f7-a982-86fa10a0e106">

Thanks!

**Comment by Jinghao-Chen** [comment]
Hi, I hope the following answers could help you
1) The reads mapped to exons of "mRNA" terms would be calculated
2) If a gene is expected to be annotated successfully, the corresponding 'gene', 'mRNA', and 'exon' line information should be included in the GFF file. And their affiliation needs to be clear (through the 'Parent' field) as shown in the following image. In addition, you could use `SAW checkGTF` to check whether the format is correct.
![a08d131a4ad9f32f386b8ecaa0590a0](https://github.com/STOmics/SAW/assets/58293762/e86c82de-a70d-4ad7-aff7-a589d9304eda)
3) I noticed that you are using SAW v7.0, which only uses gene names in the final gene expression matrix. In the latest SAW v8.0 (https://www.stomics.tech/products/BioinfoTools/OfflineSoftware), the expression matrix will include gene name and gene id.

**Comment by qiuyixmm** [comment]
It really helps me a lot. Thank you!
